pkg/columns/formatter/textcolumns/scaler.go

```diff
@@ -32,81 +32,161 @@ func (tf *TextColumnsFormatter[T]) RecalculateWidths(maxWidth int, force bool) {
+		totalAdjustedWidthNotFixed = 0
+		for _, column := range tf.showColumns {
+			if (column.col.FixedWidth || column.treatAsFixed) && !force {
+				if column.col.FixedWidth {
+					column.calculatedWidth = column.col.Width
+				}
+				continue
+			}
+
+			// set calculatedWidth based on the "weight" (relative width to other columns) of this column
+			column.calculatedWidth = int(math.Floor(float64(column.col.Width) / float64(totalWidthNotFixed) * float64(maxWidth-totalWidthFixed)))
+
+			// honor min/max widths; they'll now be treated as fixed width, afterwards we'll need another pass
+			if !force {
+				if column.col.MaxWidth > 0 && column.calculatedWidth > column.col.MaxWidth {
+					column.calculatedWidth = column.col.MaxWidth
+					column.treatAsFixed = true
+					satisfied = false
+
+					addToFixed += column.calculatedWidth
+					removeFromNotFixed += column.col.Width
+					continue
+				}
+				if column.col.MinWidth > 0 && column.calculatedWidth < column.col.MinWidth {
+					column.calculatedWidth = column.col.MinWidth
+					column.treatAsFixed = true
+					satisfied = false
+
+					addToFixed += column.calculatedWidth
+					removeFromNotFixed += column.col.Width
+					continue
+				}
+			}
+			totalAdjustedWidthNotFixed += column.calculatedWidth
 		}
-		occurrences[column.col.Name]++
-		column.calculatedWidth = int(math.Floor(float64(column.col.Width) / float64(totalWidth) * float64(maxWidth-spaces)))
-		totalAdjustedWidth += column.calculatedWidth
+
+		if satisfied {
+			break
+		}
+		totalWidthFixed += addToFixed
+		totalWidthNotFixed -= removeFromNotFixed
 	}
 
-	// Handle leftover space
-	leftover := maxWidth - (totalAdjustedWidth + spaces)
+	// Handle leftover space (gets distributed amongst non-fixed columns)
+	leftover := maxWidth - (totalAdjustedWidthNotFixed + totalWidthFixed)
 	for {
-		if leftover == 0 {
+		if leftover <= 0 {
 			break
 		}
 
+		// keep track whether we actually got to distribute space (e.g. can't do that if all columns are fixed)
 		spent := false
 
-		// distribute
+		alreadySpent := make(map[string]struct{})
+
+		// distribute one to each remaining candidate
 		for _, column := range tf.showColumns {
-			if column.col.FixedWidth && !force {
+			if (column.col.FixedWidth || column.treatAsFixed) && !force {
 				continue
 			}
-			if occurrences[column.col.Name] > 1 {
+
+			// we can only distribute to columns that are used more than once if we have leftover space for all
+			// occurrences
+			if occ := occurrences[column.col.Name]; occ > 1 {
+				if _, ok := alreadySpent[column.col.Name]; ok {
+					// we already distributed to this column in this pass (on another occurrence)
+					continue
+				}
+				if occ <= leftover {
+					column.calculatedWidth += 1
+					leftover -= occ
+					spent = true
+
+					if leftover == 0 {
+						break
+					}
+
+					alreadySpent[column.col.Name] = struct{}{}
+					continue
+				}
 				// cannot redistribute here, since it would be used more than just once
 				continue
```

Comment:
Not exactly: the columns and their calculated widths are only referenced using pointers - so if a column gets used twice, we need to make sure we have two leftover characters, since increasing the width by 1 for that column will also affect the output twice. And to make sure that in a pass a column doesn't get double that amount (but is distributed evenly), we mark that column.